src/deploy.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2022 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Deploys model to Vertex AI Prediciton
+
+Deploys model to Vertex AI Prediction
+"""
+import google.cloud.aiplatform as aip
+import gcsfs
+import json
+import os
+from absl import app
+from absl import flags
+from absl import logging
+from absl.flags import argparse_flags
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('project', None, 'Deployment project ID')
+flags.DEFINE_string('model_display_name', None, 'Name of the deployed model')
+flags.DEFINE_string('serving_container_image_uri', None, 'Serving containter image URI')
+flags.DEFINE_string('model_path', None, 'GSC path of model to deploy')
+flags.DEFINE_string('machine_type', None, 'Deployment machine type')
+flags.DEFINE_string('gpu_type', None, 'Deployment GPU type')
+flags.DEFINE_integer('gpu_count', 1, 'Number of deployment GPUs')
+flags.DEFINE_string('endpoint_name', None, 'Name of deployment endpoint')
+flags.DEFINE_string('region', None, 'Deployment region')
+
+def main(argv):
+  del argv
+
+  endpoint_name = FLAGS.endpoint_name
+  if not endpoint_name:
+    endpoint_name = FLAGS.model_display_name
+
+  existing_endpoints = aip.Endpoint.list(
+      project=FLAGS.project,
+      order_by="create_time",
+      filter='display_name="{}"'.format(endpoint_name),
```

Comment:
Hey I'm not familiar w/ this API, but I was reading through https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform.Endpoint#google_cloud_aiplatform_Endpoint_list, and using display_name seems a little brittle, since it's not unique.

What if the user explicitly specifies a `endpoint_id` value (or have some default)?  This would also simplify the instructions, so the user doesn't need to inspect the output to find out what was the endpoint_id of the endpoint that was created.

Reply:
As your code is right now, when the Endpoint.create method is called, the endpoint_id is not being passed in; as a result, Vertex will auto-generate an endpoint_id for you.  That's what the documentation says: https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform.Endpoint#google_cloud_aiplatform_Endpoint_create

My suggestion here is that rather than having Vertex auto-generate an endpoint_id, you could pass this in directly.  Thus, when you later on need to lookup an endpoint,  you can do so using the endpoint_id.  It's much like how you lookup by display_name, except that it's more robust.

> Using the endpoint_id is a good feature, but the user will have to manually look it up which adds friction.
That's exactly the opposite of what I'm saying.  In fact, right now the README instructions require the user lookup the endpoint_id.  But if the endpoint_id is specified in the create method, then this would not be necessary.

In any case my comment is not blocking.  It's a suggestion for an improvement.

Reply:
Updated.